core/client/fs/src/main/java/alluxio/client/file/ufs/UfsFileInStream.java

```diff
@@ -79,17 +85,11 @@ public int read(byte[] b, int off, int len) throws IOException {
       return -1;
     }
     updateStreamIfNeeded();
-    int currentRead = 0;
-    int totalRead = 0;
-    while (totalRead < len) {
-      currentRead = mUfsInStream.get().read(b, off + totalRead, len - totalRead);
-      if (currentRead <= 0) {
-        break;
-      }
-      totalRead += currentRead;
+    int bytesRead = mUfsInStream.get().read(b, off, len);
+    if (bytesRead > 0) {
+      mPosition += bytesRead;
```

Comment:
This "try-to-read-as-much-as-possible" while-block is moved to underfs/s3a, right?
What if we are using some other underfs?